src/plugins/navigation_embeddable/public/components/editor/navigation_embeddable_link_editor.tsx

```diff
@@ -147,7 +151,11 @@ export const NavigationEmbeddableLinkEditor = ({
             />
           </EuiFormRow>
 
-          <EuiFormRow label={NavEmbeddableStrings.editor.linkEditor.getLinkDestinationLabel()}>
+          <EuiFormRow
+            label={NavEmbeddableStrings.editor.linkEditor.getLinkDestinationLabel()}
+            isInvalid={Boolean(destinationError)}
+            error={destinationError}
+          >
```

Comment:
Only external links currently have validation - you cannot select an invalid dashboard and, even if the dashboard is deleted and you are editing the link, we simply treat this as "no selection" rather than "invalid selection" since the panel editor flyout already tells the user that the dashboard was deleted - we don't need to duplicate this information.